src/assets/scripts/client/aircraft/Pilot/Pilot.js

```diff
@@ -289,41 +288,22 @@ export default class Pilot {
      *
      * @for Pilot
      * @method applyDepartureProcedure
-     * @param procedureId {String}          the identifier for the procedure
+     * @param routeString {String}          the route
      * @param airportIcao {string}          airport icao identifier
      * @return {array}                      [success of operation, readback]
      */
-    applyDepartureProcedure(procedureId, airportIcao) {
-        const procedureModel = NavigationLibrary.getProcedure(procedureId);
+    applyDepartureProcedure(routeString, airportIcao) {
+        const [successful, response] = this._fms.replaceDepartureProcedure(routeString, airportIcao);
 
-        if (_isNil(procedureModel)) {
-            return [false, 'SID name not understood'];
-        }
-
-        if (!runwayModel) {
-            return [false, 'unsure if we can accept that procedure; we don\'t have a runway assignment'];
-        }
-
-        const exitName = procedureModel.getRandomExitPoint();
-        const routeString = `${runwayModel.name}.${procedureId}.${exitName}`;
-
-        if (!procedureModel.hasEntry(runwayModel.name)) {
-            return [
-                false,
-                `unable, the ${procedureModel.name.toUpperCase()} departure not valid ` +
-                `from Runway ${runwayModel.name.toUpperCase()}`
-            ];
+        if (!successful) {
+            return [false, response];
         }
 
         this.hasDepartureClearance = true;
 
-        this._mcp.setAltitudeVnav();
-        this._mcp.setSpeedVnav();
-        this._fms.replaceDepartureProcedure(routeString);
-
         const readback = {};
-        readback.log = `cleared to destination via the ${procedureId} departure, then as filed`;
-        readback.say = `cleared to destination via the ${procedureModel.name} departure, then as filed`;
+        readback.log = `cleared to destination via the ${this._fms._routeModel.getSidIcao().toUpperCase()} departure, then as filed`;
```

Comment:
please set `this._fms._routeModel.getSidIcao().toUpperCase()` to a constant first above the `readback` initialization.

Reply:
done